mdx/main.py

```diff
@@ -668,9 +675,22 @@ def process_mdx_cumulus(self):
         [cumulus_granules_meta.pop(ele, False) for ele in ['granuleId', 'files']]
         self.input = []
         for granule in granules:
-            for _file in granule['files']:
-                if 'metadata' not in _file.get('type', ""):
-                    self.input.append(f"s3://{_file['bucket']}/{_file['key']}")
+            files = granule['files']
+            i = 0
+            while i < len(files):
+                file = files[i]
+                if file.get('fileName', '').endswith('.cmr.json'):
```

Comment:
- Should this statement be wrapped in a `published` flag == true are we expecting the new processing approach to remove these regardless of whether or not it has the `published` flag set?
- This line should be updated to also include checking for files which end in `.cmr.xml` as that's the previous format metadata was created for and may still exist in PROD.

Reply:
1. If you don't remove any existing file records you end up with multiple cmr.json files in the file list. This is also how the dmrpp generator works: https://github.com/ghrcdaac/dmrpp-generator/blob/master/dmrpp_generator/main.py#L214
2. Since we do not generate cmr.xml and we would not go back and do this I don't think the update is needed for cmr.xml.